### DISCUSSION TITLE
Keycloak's Impersonate the user API is not working

### DISCUSSION THREAD

**Discussion by ankur-tj** [keycloak/keycloak#32768] [Q&A]
I am encountering an issue with Keycloak's Impersonate the User API where it does not function as expected. Despite setting up the impersonation role correctly and configuring the API call as per the documentation, the response leads to a sign-in page instead of directly creating an impersonation session. The correct roles are assigned, and the API call successfully returns a redirection URL, but navigating to this URL prompts a login page instead of accessing the impersonated user's account.

Key Points:

Role Assignment: Impersonation role assigned to 'customer admin'.
API Call: Correct headers and token used for the POST request.
Client Configuration: Redirection URI configured in the Keycloak client.
Observed Behavior: Redirects to a login page instead of impersonating the user directly.
For detailed information about the request configurations and responses, please refer to the full question on Stack Overflow:

https://stackoverflow.com/questions/78969173/keycloaks-impersonate-the-user-api-is-not-working @abbra @wuwx @eppsilon @cvasilak  @AlekSi

**Comment by codespearhead** [comment]
Duplicate of [1]. Could you please close this discussion [2]?

Make sure to explain your case there and upvote the issue by adding a 👍 to its description to help increase its visibility.

[1] https://github.com/keycloak/keycloak/issues/16934
[2] https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer
